compiler/rustc_infer/src/infer/mod.rs

```diff
@@ -1370,6 +1370,11 @@ impl<'tcx> InferCtxt<'tcx> {
         self.inner.borrow_mut().effect_unification_table().find(var).vid
     }
 
+    /// Check variables are equal
+    pub fn check_ty_vars_equal(&self, a: ty::TyVid, b: ty::TyVid) -> bool {
```

Comment:
Please don't make this into a separate function. It seems very easy to misuse

Reply:
Makes sense, removed.